nixio/pycore/block.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016, German Neuroinformatics Node (G-Node)
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted under the terms of the BSD License. See
+# LICENSE file in the root of the Project.
+
+from __future__ import absolute_import
+from .entity import EntityWithMetadata
+from . import util
+from . import exceptions
+from . import Group, DataArray, MultiTag, Tag, Source
+
+
+class Block(EntityWithMetadata):
+
+    def __init__(self, h5obj):
+        super(Block, self).__init__(h5obj)
+        # TODO: Validation for containers
+
+    @classmethod
+    def _create_new(cls, parent, name, type_):
+        newentity = super(Block, cls)._create_new(parent, name, type_)
+        # TODO: Create child containers
```

Comment:
In the NIX C++ library we create such subgroups only on demand. But it is not a format violation to create empty ones. 

It doesn't has to be implemented now, but it would be cool to have this feature in the pure python implementation too. You should either open an issue for this feature or implement something like the [optGroup](https://github.com/G-Node/nix/blob/master/backend/hdf5/h5x/H5Group.hpp#L237) now.